src/main/java/com/mraof/minestuck/fluid/MSFluids.java

```diff
@@ -27,25 +27,25 @@
-	public static final ForgeFlowingFluid.Properties WATER_COLORS_PROPERTIES = new ForgeFlowingFluid.Properties(WATER_COLORS, FLOWING_WATER_COLORS, FluidAttributes.builder(STILL_WATER_COLORS_TEXTURE, FLOWING_WATER_COLORS_TEXTURE)).canMultiply().bucket(() -> MSItems.WATER_COLORS_BUCKET).block(() -> MSBlocks.WATER_COLORS.get()).explosionResistance(100F);
-	public static final ForgeFlowingFluid.Properties ENDER_PROPERTIES = new ForgeFlowingFluid.Properties(ENDER, FLOWING_ENDER, FluidAttributes.builder(STILL_ENDER_TEXTURE, FLOWING_ENDER_TEXTURE)).bucket(() -> MSItems.ENDER_BUCKET).block(() -> MSBlocks.ENDER.get()).slopeFindDistance(2).levelDecreasePerBlock(2).explosionResistance(100F);
+	public static final ForgeFlowingFluid.Properties OIL_PROPERTIES = new ForgeFlowingFluid.Properties(OIL, FLOWING_OIL, FluidAttributes.builder(STILL_OIL_TEXTURE, FLOWING_OIL_TEXTURE)).bucket(() -> MSItems.OIL_BUCKET.get()).block(() -> MSBlocks.OIL.get()).slopeFindDistance(3).explosionResistance(100F);
+	public static final ForgeFlowingFluid.Properties LIGHT_WATER_PROPERTIES = new ForgeFlowingFluid.Properties(LIGHT_WATER, FLOWING_LIGHT_WATER, FluidAttributes.builder(STILL_LIGHT_WATER_TEXTURE, FLOWING_LIGHT_WATER_TEXTURE)).canMultiply().bucket(() -> MSItems.LIGHT_WATER_BUCKET.get()).block(() -> MSBlocks.LIGHT_WATER.get()).explosionResistance(100F);
+	public static final ForgeFlowingFluid.Properties BLOOD_PROPERTIES = new ForgeFlowingFluid.Properties(BLOOD, FLOWING_BLOOD, FluidAttributes.builder(STILL_BLOOD_TEXTURE, FLOWING_BLOOD_TEXTURE)).bucket(() -> MSItems.BLOOD_BUCKET.get()).block(() -> MSBlocks.BLOOD.get()).explosionResistance(100F);
+	public static final ForgeFlowingFluid.Properties BRAIN_JUICE_PROPERTIES = new ForgeFlowingFluid.Properties(BRAIN_JUICE, FLOWING_BRAIN_JUICE, FluidAttributes.builder(STILL_BRAIN_JUICE_TEXTURE, FLOWING_BRAIN_JUICE_TEXTURE)).bucket(() -> MSItems.BRAIN_JUICE_BUCKET.get()).block(() -> MSBlocks.BRAIN_JUICE.get()).explosionResistance(100F);
+	public static final ForgeFlowingFluid.Properties WATER_COLORS_PROPERTIES = new ForgeFlowingFluid.Properties(WATER_COLORS, FLOWING_WATER_COLORS, FluidAttributes.builder(STILL_WATER_COLORS_TEXTURE, FLOWING_WATER_COLORS_TEXTURE)).canMultiply().bucket(() -> MSItems.WATER_COLORS_BUCKET.get()).block(() -> MSBlocks.WATER_COLORS.get()).explosionResistance(100F);
+	public static final ForgeFlowingFluid.Properties ENDER_PROPERTIES = new ForgeFlowingFluid.Properties(ENDER, FLOWING_ENDER, FluidAttributes.builder(STILL_ENDER_TEXTURE, FLOWING_ENDER_TEXTURE)).bucket(() -> MSItems.ENDER_BUCKET.get()).block(() -> MSBlocks.ENDER.get()).slopeFindDistance(2).levelDecreasePerBlock(2).explosionResistance(100F);
```

Comment:
Here are some other functions that take suppliers. Can you do the same thing with `block()` and `bucket()` that you did with grist candy?

Reply:
finished!